data/src/isupport.rs

```diff
@@ -0,0 +1,614 @@
+            Parameter::MONITOR(_) => "MONITOR",
+            Parameter::MSGREFTYPES(_) => "MSGREFTYPES",
+            Parameter::NAMESX => "NAMESX",
+            Parameter::NETWORK(_) => "NETWORK",
+            Parameter::NICKLEN(_) => "NICKLEN",
+            Parameter::OVERRIDE => "OVERRIDE",
+            Parameter::PREFIX(_) => "PREFIX",
+            Parameter::SAFELIST => "SAFELIST",
+            Parameter::SECURELIST => "SECURELIST",
+            Parameter::SILENCE(_) => "SILENCE",
+            Parameter::STATUSMSG(_) => "STATUSMSG",
+            Parameter::TARGMAX(_) => "TARGMAX",
+            Parameter::TOPICLEN(_) => "TOPICLEN",
+            Parameter::UHNAMES => "UHNAMES",
+            Parameter::USERIP => "USERIP",
+            Parameter::USERLEN(_) => "USERLEN",
+            Parameter::UTF8ONLY => "UTF8ONLY",
+            Parameter::VLIST(_) => "VLIST",
+            Parameter::WATCH(_) => "WATCH",
+            Parameter::WHOX => "WHOX",
+            Parameter::Negation(key) => key.as_ref(),
+        }
+    }
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Debug)]
+pub enum CaseMap {
+    ASCII,
+    RFC1459,
+    RFC1459_STRICT,
+    RFC7613,
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct ChannelLimit {
+    prefix: String,
+    limit: Option<u16>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct ChannelMode {
+    letter: char,
+    modes: String,
+}
+
+#[derive(Debug)]
+pub enum ClientOnlyTags {
+    Allowed(String),
+    Denied(String),
+    DenyAll,
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct CommandTargetLimit {
+    command: String,
+    limit: Option<u16>,
+}
+
+#[derive(Debug)]
+pub enum MessageReferenceType {
+    Timestamp,
+    MessageID,
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct ModesLimit {
+    modes: String,
+    limit: u16,
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct PrefixMap {
+    prefix: char,
+    mode: char,
+}
+
+pub fn default_ban_exception_channel_letter() -> char {
+    'e'
+}
+
+pub fn default_caller_id_letter() -> char {
+    'g'
+}
+
+pub fn default_deaf_letter() -> char {
+    'D'
+}
+
+pub fn default_invite_exception_letter() -> char {
+    'I'
+}
+
+fn parse_optional_positive_integer(value: &str) -> Result<Option<u16>, &'static str> {
+    if value.is_empty() {
+        Ok(None)
+    } else if let Ok(value) = value.parse::<u16>() {
+        Ok(Some(value))
+    } else {
+        Err("optional value must be a positive integer if specified")
+    }
+}
+
+fn parse_optional_string(value: &str) -> Option<String> {
+    if value.is_empty() {
+        None
+    } else {
+        Some(value.to_string())
+    }
+}
+
+fn parse_required_letter(value: &str, default_value: Option<char>) -> Result<char, &'static str> {
+    if let Some(value) = value.chars().next() {
+        Ok(value)
+    } else if let Some(default_value) = default_value {
+        Ok(default_value)
+    } else {
+        Err("value required to be a letter")
```

Comment:
This error message doesn't line up w/ the logic: "required to be a letter" but we are actually checking it's a valid utf8 (implied by str) >1 char string.

Reply:
Good point!  Updated to require an ASCII alphabetic.